### ISSUE TITLE
Test suite failure with GHC 8

### ISSUE THREAD

**Issue by snoyberg** [thoughtpolice/hs-ed25519#15]
This is due to hlint failing, I'm not sure if it's related to GHC 8:

```
> /tmp/stackage-build8/ed25519-0.0.5.0$ runghc -clear-package-db -global-package-db -package-db=/home/stackage/work/builds/nightly/pkgdb Setup configure --enable-tests --package-db=clear --package-db=global --package-db=/home/stackage/work/builds/nightly/pkgdb --libdir=/home/stackage/work/builds/nightly/lib --bindir=/home/stackage/work/builds/nightly/bin --datadir=/home/stackage/work/builds/nightly/share --libexecdir=/home/stackage/work/builds/nightly/libexec --sysconfdir=/home/stackage/work/builds/nightly/etc --docdir=/home/stackage/work/builds/nightly/doc/ed25519-0.0.5.0 --htmldir=/home/stackage/work/builds/nightly/doc/ed25519-0.0.5.0 --haddockdir=/home/stackage/work/builds/nightly/doc/ed25519-0.0.5.0 --flags=
Configuring ed25519-0.0.5.0...
> /tmp/stackage-build8/ed25519-0.0.5.0$ runghc -clear-package-db -global-package-db -package-db=/home/stackage/work/builds/nightly/pkgdb Setup build
Building ed25519-0.0.5.0...
Preprocessing library ed25519-0.0.5.0...
[1 of 1] Compiling Crypto.Sign.Ed25519 ( src/Crypto/Sign/Ed25519.hs, dist/build/Crypto/Sign/Ed25519.o )
Preprocessing test suite 'properties' for ed25519-0.0.5.0...
[1 of 1] Compiling Main             ( tests/properties.hs, dist/build/properties/properties-tmp/Main.o )
Linking dist/build/properties/properties ...
Preprocessing test suite 'hlint' for ed25519-0.0.5.0...
[1 of 1] Compiling Main             ( tests/hlint.hs, dist/build/hlint/hlint-tmp/Main.o )
Linking dist/build/hlint/hlint ...
Preprocessing test suite 'doctests' for ed25519-0.0.5.0...
[1 of 1] Compiling Main             ( tests/doctests.hs, dist/build/doctests/doctests-tmp/Main.o )
Linking dist/build/doctests/doctests ...
> /tmp/stackage-build8/ed25519-0.0.5.0$ dist/build/properties/properties
Signature roundtrip                     : +++ OK, passed 100 tests.
Detached signature roundtrip            : +++ OK, passed 100 tests.
Detached signature length               : +++ OK, passed 100 tests.
Detached signature length (#2)          : +++ OK, passed 100 tests.
Passed 400 tests!
> /tmp/stackage-build8/ed25519-0.0.5.0$ dist/build/hlint/hlint
src/Crypto/Sign/Ed25519.hs:391:1: Warning: Eta reduce
Found:
  sign' sk xs = dsign sk xs
Why not:
  sign' = dsign

src/Crypto/Sign/Ed25519.hs:406:1: Warning: Eta reduce
Found:
  verify' pk xs sig = dverify pk xs sig
Why not:
  verify' = dverify

2 hints
```

**Comment by thoughtpolice** [comment]
Oi, strange. I thought I told [hlint to ignore eta reduction warnings](https://github.com/thoughtpolice/hs-ed25519/blob/master/HLint.hs), though.

I'll take a look. I need to do a release of 0.6.0.0 with newer documentation and fix up the CI to include 8.0.1, etc.

**Comment by thoughtpolice** [comment]
I haven't seen this with GHC 8.0.1 or anything, and as I said, I told HLint to ignore eta reduction warnings, so maybe this is just a random ghost?

Closing, due to inability to reproduce.
